Sources/TestingMacros/ConditionMacro.swift

```diff
@@ -293,6 +293,26 @@ public struct AmbiguousRequireMacro: RefinedConditionMacro {
   }
 }
 
+/// A type describing the expansion of the `#require()` macro when it is passed
+/// a non-optional, non-`Bool` value.
+///
+/// This type is otherwise exactly equivalent to ``RequireMacro``.
+public struct NonOptionalRequireMacro: RefinedConditionMacro {
+  public typealias Base = RequireMacro
+
+  public static func expansion(
+    of macro: some FreestandingMacroExpansionSyntax,
+    in context: some MacroExpansionContext
+  ) throws -> ExprSyntax {
+    if let argument = macro.arguments.first {
```

Comment:
Where does this check if it is a Bool?

Reply:
I see, you are trying to not show a warning for a bool value, which this code does not do

Reply:
`Bool` and `Bool?` won't hit this particular macro variant, they'll hit the normal one (or the "ambiguous `Bool?`" one.)